src/firmware/app/world/firmware_robot.c

```diff
@@ -3,6 +3,41 @@
 #include <math.h>
 #include <stdlib.h>
 
+// these are set to decouple the 3 axis from each other
+// the idea is to clamp the maximum velocity and acceleration
+// so that the axes would never have to compete for resources
+#define TIME_HORIZON 0.05f  // s
+
+typedef void (*TrajectoryFollower_t)(const FirmwareRobot_t*, PositionTrajectory_t, size_t,
+                                     float);
+typedef void (*ApplyDirectPerWheelPower_t)(
+    const FirmwareRobot_t*, TbotsProto_DirectControlPrimitive_DirectPerWheelControl);
+typedef void (*SetLocalVelocity_t)(
+    const FirmwareRobot_t*, TbotsProto_DirectControlPrimitive_DirectVelocityControl);
+typedef void (*StopRobot_t)(const FirmwareRobot_t*, TbotsProto_StopPrimitive_StopType);
```

Comment:
Would be nice to have some comments here explaining what each function does

Reply:
Done.